libqtile/widget/sensors.py

```diff
@@ -81,32 +73,19 @@ def __init__(self, **config):
                 break
 
     def get_temp_sensors(self):
-        """calls the unix `sensors` command with `-f` flag if user has specified that
-        the output should be read in Fahrenheit.
         """
-        command = ["sensors", ]
-        if not self.metric:
-            command.append("-f")
-        try:
-            sensors_out = self.call_process(command)
-            if not sensors_out:
-                return None
-        except FileNotFoundError:
-            return None
-        except CalledProcessError:
-            return ""
-        return self._format_sensors_output(sensors_out)
-
-    def _format_sensors_output(self, sensors_out):
-        """formats output of unix `sensors` command into a dict of
-        {<sensor_name>: (<temperature>, <temperature symbol>), ..etc..}
+        Reads temperatures from sys-fs via psutil.
+        Output will be read Fahrenheit if user has specified it to be.
         """
-        temperature_values = {}
-        logger.debug(self.sensors_temp.findall(sensors_out))
-        for name, temp, symbol in self.sensors_temp.findall(sensors_out):
-            name = name.strip()
-            temperature_values[name] = temp, symbol
-        return temperature_values
+
+        temperature_list = {}
+        temps = psutil.sensors_temperatures(fahrenheit=not self.metric)
+        unit = '°C' if self.metric else '°F'
+        for kernel_module in temps:
+            for sensor in temps[kernel_module]:
+                temperature_list[sensor.label] = (str(round(sensor.current, 1)), unit)
```

Comment:
On my laptop, there are multiple sensors with an empty label. This only saves one of them in the `temperature_list` dict and it's not clear which module it's attached too.

Reply:
I added an indexed named generation for this case.
The problem should have been occurring with the old implementation as well as it is using the same back-end for its values.